rust_snuba/src/strategies/commit_log.rs

```diff
@@ -81,6 +92,105 @@ impl TryFrom<Commit> for KafkaPayload {
     }
 }
 
+struct ProduceMessage {
+    producer: Arc<dyn Producer<KafkaPayload>>,
+    topic: Arc<TopicOrPartition>,
+    skip_produce: bool,
+}
+
+impl ProduceMessage {
+    #[allow(dead_code)]
+    pub fn new(
+        producer: impl Producer<KafkaPayload> + 'static,
+        topic: TopicOrPartition,
+        skip_produce: bool,
+    ) -> Self {
+        ProduceMessage {
+            producer: Arc::new(producer),
+            topic: Arc::new(topic),
+            skip_produce,
+        }
+    }
+}
+
+impl TaskRunner<KafkaPayload, KafkaPayload> for ProduceMessage {
+    fn get_task(&self, message: Message<KafkaPayload>) -> RunTaskFunc<KafkaPayload> {
+        let producer = self.producer.clone();
+        let topic = self.topic.clone();
+        let skip_produce = self.skip_produce;
+
+        Box::pin(async move {
+            if skip_produce {
+                return Ok(message);
+            }
+
+            let mut retries = 3;
+            while retries > 0 {
+                match producer.produce(&topic, message.payload()) {
+                    Ok(_) => {
+                        return Ok(message);
+                    }
+                    Err(e) => {
+                        retries -= 1;
+                        sleep(Duration::from_millis(100));
```

Comment:
Shouldn't this be a parameter ?

Reply:
Removed, this was supposed to be temporary code